Src/IronPython/Runtime/PythonEncoding.cs

```diff
@@ -856,4 +998,105 @@ protected void Throw(byte[] bytesUnknown, int index)
                 => throw new DecoderFallbackException($"'surrogatepass' error handler: not a surrogate character", bytesUnknown, index);
         }
     }
+
+    internal class PythonErrorHandlerEncoding : PythonEncoding {
```

Comment:
Maybe move this to `PythonErrorHandlerEncoding.cs`? I would suggest the same for the other classes in here but it might make reviewing a bit difficult if it were done in this PR. We could probably add an `IronPython.Runtime.Text` namespace and move all the encoding stuff in there... We could also move some of the classes in StringOps to make the file a bit lighter?

Reply:
I like the idea of `IronPython.Runtime.Text` and cleaning up `StringOps` from encoding stuff, I am just not sure the time is ripe. You may have noticed that `PythonEncoding` and friends are much more complex than strictly necessary from the current usage point of view. This is because they support incremental encoding/decoding, but IronPython runtime never uses it in this capacity. 

I developed it this way so that it could possibly evolve to [one of the solutions described earlier](https://github.com/IronLanguages/ironpython3/issues/906#issuecomment-656807221). If so, there is more to come. If not, the whole `PythonEncoding` machinery could be substantially pruned, and perhaps even merged in to one class.

In any case, I'd prefer to do namespace refactoring is a separate PR.